src/components/IconToggle/IconToggle.tsx

```diff
@@ -50,12 +73,16 @@ export class IconToggle extends React.Component<IconToggleProps> {
   };
 
   render() {
-    const { labelPlacement, onChange, checked, disabled, ...rest } = this.props;
+    const { labelPlacement, onChange, checked, disabled, animation, ...rest } = this.props;
+
+    const { animated } = this.state;
```

Comment:
I think we can have the same animation without `this.state.animated`.
Maybe we can try to apply the animation using the `animation-direction` css rule.